x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -31,56 +26,11 @@ impl IntoIterator for Address {
     type IntoIter = std::array::IntoIter<Self::Item, ADDRESS_LEN>;
 
     fn into_iter(self) -> Self::IntoIter {
-        IntoIterator::into_iter(self.0 .0)
-    }
-}
-
-/// A struct representing a fixed length of 32 bytes.
-/// This can be used for passing strings to the host. It caps the string at 32 bytes,
-#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug)]
-pub struct Bytes32([u8; Self::LEN]);
-impl Bytes32 {
-    pub const LEN: usize = 32;
-    #[must_use]
-    pub fn new(bytes: [u8; Self::LEN]) -> Self {
-        Self(bytes)
-    }
-    #[must_use]
-    pub fn as_bytes(&self) -> &[u8] {
-        &self.0
-    }
-}
-
-/// Implement the Display trait for Bytes32 so that we can print it.
-/// Enables `to_string()` on Bytes32.
-impl std::fmt::Display for Bytes32 {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        // Find the first null byte and only print up to that point.
-        let null_pos = self.0.iter().position(|&b| b == b'\0').unwrap_or(Self::LEN);
-        String::from_utf8_lossy(&self.0[..null_pos]).fmt(f)
-    }
-}
-
-impl From<String> for Bytes32 {
-    fn from(value: String) -> Self {
-        let mut bytes: [u8; Self::LEN] = [0; Self::LEN];
-        bytes[..value.len()].copy_from_slice(value.as_bytes());
-        Self(bytes)
-    }
-}
-
-impl From<i64> for Bytes32 {
-    fn from(value: i64) -> Self {
-        let bytes: [u8; Self::LEN] = unsafe {
-            // We want to copy the bytes here, since [value] represents a ptr created by the host
-            std::slice::from_raw_parts(value as *const u8, Self::LEN)
-                .try_into()
-                .unwrap()
-        };
-        Self(bytes)
+        IntoIterator::into_iter(self.0)
     }
 }
 
+/// A trait that represents an argument that can be passed to & from the host.
 pub trait Argument {
```

Comment:
only place we are using `Argument` now is when we are calling an external program. I think we can remove this trait in the future, going to mark as an issue and hit up in another PR.